pinot-core/src/main/java/org/apache/pinot/core/data/manager/TableDataManagerType.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.manager;
+
+public enum TableDataManagerType {
```

Comment:
If possible, let's not introduce this new type. One option is to add the `DIMENSION` as a `TableType`, but I'm not sure if we are going to support real-time dimTable in the future (we can add another `REALTIME_DIMENSION` maybe)

Reply:
I refrained from adding a new table type because of the following two reasons:

1. Since ingestion mode is still offline, it seems appropriate that table type still remain OFFLINE
2. Adding a new type would be an invasive change and I wasn't sure of the blast radius.

Instead of adding a new `TableDataManagerType` I have now added a boolean similar to how we are identifying the dim table in table config. Even for a future realtime dim table, this approach should work. Does this approach look good?

Reply:
Some suggestions: We could use `TableSubtype` maybe. And make it an enum so that we have some room in the future. I am not sure if, for example, a different layout of a larger table can also support joins.

Reply:
@dharakk  why was this suggestion not implemented?